### PR TITLE
Determine CPU info for ARM processors

### PR DESCRIPTION
**Pull request by djmitche**
This is loosely based on lscpu, and organizes the data to fit the
existing field defintiions and formats.

It seems that `/proc/cpuinfo` contains lots of useful detail for x86 architectures, but very little for ARM, so this fetches data from various /sys files instead, as lscpu does.

Note that CI doesn't cover ARM.  I'm not sure of an easy way to fix that!

### CODE REVIEW

**Comment by djmitche** [comment]
It'd be great if someone could run this on a rPi.

File: cpu/cpu_linux_arm64.go [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+type hwImpl struct {
+	// name of the implementer
+	name string
+	// part numbers (indexed by "CPU part")
+	parts map[uint64]string
+}
+
+// hwVariant is based on hw_implementer in
+//     https://github.com/util-linux/util-linux/blob/master/sys-utils/lscpu-arm.c
+// See from-lscpu-arm.py to regenerate this value.
```
Comment (hush-hush) [inline cpu/cpu_linux_arm64.go:31]: Should we isolate this in its own file that the python script can overwrite ?

File: cpu/from-lscpu-arm.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env python
```
Comment (hush-hush) [inline cpu/from-lscpu-arm.py:1]: Could we add something to the README about how to use this and when to update the ARM cpu definition ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

**Review by hush-hush — APPROVED** [review]
On nit on the README but looks good otherwise

**Review by sgnn7 — APPROVED** [review]
@djmitche LGTM just a couple of nits

File: cpu/util_linux.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	return result, true
+}
+
+// readProcCpuInfo reads /proc/cpuinfo.  The file is structured as a set of
+// blane-line-separated stanzas, and each stanza is a map of string to string,
+// with whitespace stripped.
+func readProcCpuInfo() ([]map[string]string, error) {
+	file, err := os.Open(testingPrefix + "/proc/cpuinfo")
```
[the remark this answers is not in the document]
Comment (djmitche) [reply cpu/util_linux.go:104]: What non-testing context are you seeing it in?

**Comment by djmitche** [comment]
That should fix the CPU issue -- can you test again?

**Comment by djmitche** [comment]
(and, it seems that Linux does not know the cache sizes on rPi, so that's the best we can do)

File: cpu/util_linux.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sgnn7) [reply cpu/util_linux.go:106]: The same code is run when you regularly build the `gohai` executable or import this as a module. The `testing` prefix on this var is much less fitting for those contexts.

**Comment by sgnn7** [comment]
@djmitche Looks good!

```json
{
  "cache_size": "0 KB",
  "cache_size_l1": "0",
  "cache_size_l2": "0",
  "cache_size_l3": "0",
  "cpu_cores": "4",
  "cpu_logical_processors": "4",
  "cpu_numa_nodes": "0",
  "cpu_pkgs": "1",
  "family": "none",
  "model": "0xd08",
  "model_name": "Cortex-A72",
  "stepping": "r0p3",
  "vendor_id": "ARM"
}
```

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**